src/main/java/gyro/google/dns/ManagedZoneResource.java

```diff
@@ -0,0 +1,405 @@
+        this.dnssecConfig = dnssecConfig;
+    }
+
+    /**
+     * The presence for this field indicates that outbound forwarding is enabled for this zone. The value of this field contains the set of destinations to forward to.
+     *
+     * @subresource gyro.google.dns.ZoneForwardingConfig
+     */
+    @ConflictsWith("dnssec-config")
+    // XXX: https://github.com/perfectsense/gyro/issues/190
+    @Updatable
+    public ZoneForwardingConfig getForwardingConfig() {
+        return forwardingConfig;
+    }
+
+    public void setForwardingConfig(ZoneForwardingConfig forwardingConfig) {
+        this.forwardingConfig = forwardingConfig;
+    }
+
+    /**
+     * User labels.
+     */
+    @Updatable
+    public Map<String, String> getLabels() {
+        if (labels == null) {
+            labels = new HashMap<>();
+        }
+        return labels;
+    }
+
+    public void setLabels(Map<String, String> labels) {
+        this.labels = labels;
+    }
+
+    /**
+     * User assigned name for this resource. Must be unique within the project. The name must be 1-63 characters long, must begin with a letter, end with a letter or digit, and only contain lowercase letters, digits or dashes.
+     */
+    @Required
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Optionally specifies the NameServerSet for this ManagedZone. A NameServerSet is a set of DNS name servers that all host the same ManagedZones. Most users will leave this field unset.
+     */
+    public String getNameServerSet() {
+        return nameServerSet;
+    }
+
+    public void setNameServerSet(String nameServerSet) {
+        this.nameServerSet = nameServerSet;
+    }
+
+    /**
+     * Delegate your managed_zone to these virtual name servers.
+     */
+    @Output
+    public List<String> getNameServers() {
+        if (nameServers == null) {
+            nameServers = new ArrayList<>();
+        }
+        return nameServers;
+    }
+
+    public void setNameServers(List<String> nameServers) {
+        this.nameServers = nameServers;
+    }
+
+    /**
+     * For privately visible zones, the set of Virtual Private Cloud resources that the zone is visible from.
+     *
+     * @subresource gyro.google.dns.ZonePrivateVisibilityConfig
+     */
+    @ConflictsWith("dnssec-config")
+    // XXX: https://github.com/perfectsense/gyro/issues/190
+    @Updatable
+    public ZonePrivateVisibilityConfig getPrivateVisibilityConfig() {
+        return privateVisibilityConfig;
+    }
+
+    public void setPrivateVisibilityConfig(ZonePrivateVisibilityConfig privateVisibilityConfig) {
+        this.privateVisibilityConfig = privateVisibilityConfig;
+    }
+
+    /**
+     * The zone's visibility: public zones are exposed to the Internet, while private zones are visible only to Virtual Private Cloud resources.
+     */
+    @ValidStrings({ "private", "public" })
```

Comment:
Shouldn't the visibility be updatable? For example if I create a managed zone as private with a private visibility config. But then update it by removing that config and setting to "public" this operation is skipped when using `gyro up`.

Reply:
No, it's not allowed.